src/ast/operator.rs

```diff
@@ -86,41 +88,59 @@ pub enum BinaryOperator {
     PGRegexIMatch,
     PGRegexNotMatch,
     PGRegexNotIMatch,
+    PGCustomBinaryOperator(PGCustomOperator),
+}
+
+/// PostgreSQL-specific custom operator.
+///
+/// See [CREATE OPERATOR](https://www.postgresql.org/docs/current/sql-createoperator.html) for more information.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct PGCustomOperator {
+    pub schema: Option<String>,
+    pub name: String,
```

Comment:
What would you think about using the pre-existing `Identifier` here (that supports various qualified names)

like:

```suggestion
pub struct PGCustomOperator {
    pub ident: ObjectName
```

Thank you could use `parse_object_name` instead of custom parsing logic. 

https://github.com/sqlparser-rs/sqlparser-rs/blob/076b587bb26f725a9a172c2ca4da373e83937158/src/parser.rs#L3114-L3125

This would likely require less code as well as handle parsing more cases (like `OPERATOR("~")` -- aka quoted strings)

Reply:
I tried doing this, but looks like `Identifier` doesn't allow for `Token`s that are un-quoted operators, for e.g: `~` is not a valid identifier. I am not sure if extending the `parse_identifier` to allow for un-quoted operators is the right call here.

IMO we could start with the existing mechanism of schema qualifier operator name (though I would've liked to use the `parse_object_name` logic) and can extend it as needed later. Let me know.